mbed_lstools/lstools_base.py

```diff
@@ -323,20 +337,8 @@ def write_mock_file(filename, mock_ids):
                     str(e)))
             return False
 
-        lock = self.mbedls_get_global_lock()
-        while not lock.i_am_locking():
-            try:
-                lock.acquire(timeout=1)
-            except LockTimeout as e:
-                lock.break_lock()
-                lock.acquire()
-                self.debug(self.mock_read.__name__, str(e))
-        self.debug(self.mock_write.__name__, "locked '%s'"% lock.path)
-
         result = write_mock_file(self.MOCK_HOME_FILE_NAME, mock_ids)
```

Comment:
Any reason why the function `write_mock_file` can't be inlined here? I suppose this was mostly keep it parallel to how `mock_read` is written?